pkg/kubelet/kubelet_node_status.go

```diff
@@ -46,11 +46,15 @@ import (
 )
 
 // registerWithAPIServer registers the node with the cluster master. It is safe
-// to call multiple times, but not concurrently (kl.registrationCompleted is
-// not locked).
+// to call multiple times.
 func (kl *Kubelet) registerWithAPIServer() {
-	if kl.registrationCompleted {
-		return
+	// If the informer has not synced try to register to improve the node startup latency
```

Comment:
this makes a difference, registration is much faster not waiting for the informer and is not really a problem , if the node still exists the subsequent Create() call will fail